testing/file_test/autoupdate.cpp

```diff
@@ -265,6 +333,8 @@ auto AutoupdateFileTest(
       // output_line_number.
       if (non_check_line.line_number() < 1) {
         new_lines.push_back(&non_check_line);
+        // These are ignored when calculated line counts.
+        ++file_offset_in_new_lines;
```

Comment:
I'd find this easier to reason about with a `CHECK(file_offset_in_new_lines == new_lines.size());` after this line -- I think we expect all of these lines to be at the start of the output for the file, and they aren't part of the file contents.

Reply:
Adding to #3214